### ISSUE TITLE
better indicators for bot and system messages

### ISSUE THREAD

**Issue by cyphar** [kennib/necsus#42]
Right now all messages look like they came from users, but bot and system messages really should get some special styling (maybe a `[bot]` tag like we have on GitHub). System messages are particularly odd because they look like they came from a user called `necsus`. Simplest solution might be to just add a "role" column to the messages table (which can't be set by users directly).

The slightly complication is how we'll handle bots which make use of the messaging API directly (rather than just returning JSON). Maybe we should have a parameter that indicates the message is from a user (which defaults to off), so student bot messages will be treated as bots by default?

**Comment by kennib** [comment]
I don't really mind if bots are not distinguished. The students can choose to name their bots anything as it is. That is, they at least have the option to name them as X Bot at the moment.

What I'm more concerned about is that with the proposed async bots that there is no feedback on the frontend that a bot is currently being triggered. A spinning wheel in the sidebar next to a bot while the server is waiting for a response is the kind of thing I'm imagining.

To understand why I think this would be helpful imagine a scenario where the URL for a bot is slightly wrong. The students sends a message. The student won't receive feedback in the logs on their bot server. They won't have any feedback in the Necsus client aside from the posted message. Was a bot triggered by my message or not?

An additional feature that could help this problem might be to highlight the triggered bots (in the sidebar) on hover of a message. This would involve sending back some extra info with the message JSON. I'm not sure how cumbersome this would be to implement at the moment.

**Comment by cyphar** [comment]
Yeah that wouldn't be too hard to provide, though I still would like at least the "system" errors to look less like normal messages (maybe something as simple as colouring the name red with a light red background -- or something like that).

Hover-over for "which bots were triggered by this message" regexes might be a bit annoying because the regexes are Python regexes while JS has a slightly different dialect so we'll need to do a round-trip on each hover... (Or maybe we can generate a "simplified" version of the regex internally -- (stripping out `(?P<>...)` for instance) which JS can then understand.)

**Comment by cyphar** [comment]
By the way, I've dropped the async bots precisely so we can keep this kind of "bot still thinking" behaviour. So really all this would take is to add a spinning wheel anywhere on the page until the request to the server completes.

**Comment by joelgibson** [comment]
User messages (those triggered by a POST to `/api/actions/message`) are now distinguished from bot messages (those returned by NeCSuS interacting with a bot), and from system messages, via a `kind` field on the message, which can be `user`, `bot`, or `system`. I've also made system messages show as grey in the NeCSuS client (this is partly because I added a lot more debugging messages from the system, to help the students out).

I agree with Kenni in that not visually distinguishing user and bot messages is a good default. The students can now if they really want to...
